src/server/decode.rs

```diff
@@ -86,11 +88,24 @@ where
         "Unexpected Content-Length header"
     );
 
+    let (sender, receiver) = sync::channel(1);
```

Comment:
With regards to #137, using [async-channel](https://docs.rs/async-channel) may actually be better here.

Reply:
I'm not sure how much that pr/issue actually matters because async-h1 depends on http-types, which also uses unstable